github_nonpublic_api/api.py

```diff
@@ -46,8 +51,13 @@ def _get_and_submit_form(session, url: str, data_callback=None, form_matcher=lam
     submit_url = urljoin(url, action_url)
     logging.info('Posting form to URL %s', submit_url)
 
-    response = session.post(submit_url, data=data)
-    response.raise_for_status()
+    try:
+        response = session.post(submit_url, data=data)
+        response.raise_for_status()
+        logging.info('Form submitted successfully to URL %s', submit_url)
+    except requests.exceptions.RequestException as e:
+        logging.error('Error submitting form to URL %s: %s', submit_url, str(e))
+        raise
```

Comment:
same comment here as above, we want to avoid possibly double logging on the error case, so it is fine for it to just bubble up.

I also have concerns about logging the success every time at info, as that's going to generate a lot of log message with unclear value.